SETUP/DEVELOPMENT.md

```diff
@@ -0,0 +1,102 @@
+# Code Development
+
+This doc is designed for users who want to develop on the DProofreaders (DP)
+codebase.
+
+## Introduction
+
+The DP code is almost entirely server-side PHP with a smattering of JS and
+CSS. It uses a MySQL backend database and requires a web server -- pgdp.net
+uses Apache but Nginx should work as well. See the [installation docs](INSTALL.md)
+for information on minimum middleware versions.
+
+The code does not use a PHP framework or an ORM. It was developed 20 years ago
+and has grown very organically over the years. There has been effort in the
+past decade to standardize things but you'll find many different patterns remain.
+We have a defined [coding style](CODE_STYLE.md) for the codebase and a set of
+[best practices](https://www.pgdp.net/wiki/DP_Code_Best_Practices) that cover
+some of our prescribed patterns.
+
+## Contributing
+
+DP code uses git for source control and is housed at GitHub in the
+[DProofreaders/dproofreaders](https://github.com/DistributedProofreaders/dproofreaders)
+repository -- likely where you got this code to begin with.
+
+We use the "fork, branch, develop, merge" model, where developers are expected to:
+1. fork the main repository
+2. create a development branch off master
+3. develop code on the branch
+4. generate a merge/pull request when ready to merge
+
+See also [DP Code Development Using git](https://www.pgdp.net/wiki/DP_Code_Development_Using_git)
+and a similar guide specific to the [dproofreaders](https://www.pgdp.net/wiki/DP_Code_Development_Using_git:_dproofreaders)
+repo.
+
+## Development Environment
+
+The best current development environment is using the
+[DP development VM](https://www.pgdp.net/wiki/DP_Code_Development_VM) which
+comes with a recent release of the DP code, all required middleware, and some
+sample data. There is a desire to get a true local development environment --
+via vagrant or docker -- created but that has not yet been completed.
+
+### Tooling
+
+The code uses [Composer](https://getcomposer.org/) to manage PHP dependencies.
+Some of these dependencies are runtime dependencies, others are just used for
+development (phpstan, phpunit, etc). To install PHP dependencies (runtime
+and development) run the following in the repo base:
+```bash
+composer install
+```
+
+The code also uses [npm](https://www.npmjs.com/) for some _development only_
```

Comment:
weirdly I don't see a download for npm on that site. I've only ever managed to install npm through the nodejs installer. We should probably call that our here as well as which version of node. Currently I think our tasks use node 20, the current LTS.

Reply:
Are you able to provide some text for here on installing nodejs/npm? On TEST we have the Ubuntu 20.04-provided nodejs and npm version which are ancient.

Reply:
Since node/npm is only used in dev scenarios (tests and less) I don't think the installed version matters much. For development though, there are a few options depending on your OS/distro.
1. Ubuntu - I use nodesource's repositories/instructions outlined here: https://github.com/nodesource/distributions?tab=readme-ov-file#debian-and-ubuntu-based-distributions
2. Windows - download installer from https://nodejs.org

If you need to manage multiple node/npm versions, you can use tools like nvm on linux: https://github.com/nvm-sh/nvm

Note, nodejs installed npm with it. They pair versions but aren't the same. Currently node 20 pairs with npm 10.

Reply:
Thanks Chris, that was really helpful! I've updated the docs, let me know if they make more sense.

Reply:
Looks good to me!

Reply:
I used the previous version of instructions which referred to (https://nodejs.org/en/download/package-manager), which worked flawlessly for me using the VirtualBox VM.